### ISSUE TITLE
registered with client and replicated

### ISSUE THREAD

**Issue by conansoft** [hashicorp/consul#6985]
I am just started playing with consul, I have setup a 3 server,1 client cluster, the three servers joined each other, and elected a leader.

I am using this for service discovery, and registered the services by client node,but the  services are not forwarded to server by rpc ,they stored on the client node。

In addition, when I am registered the services with one of the follower servers in the cluster.However the data is not getting replicated to other servers in the cluster. 

server1:
consul agent -server -ui -bind=0.0.0.0   -data-dir /data/consul -node=server1 -client=0.0.0.0 -bootstrap-expect=3 
server2:
consul agent -server -ui   -bind=0.0.0.0 -data-dir /data/consul -node=server2  -client=0.0.0.0 -join $ip
server3:
consul agent -server -ui   -bind=0.0.0.0 -data-dir /data/consul -node=server3  -client=0.0.0.0 -join $ip
client:
consul agent  -ui   -bind=0.0.0.0 -data-dir /data/consul -node=client1  -client=0.0.0.0 -join $ip

**Comment by like-inspur** [comment]
yes, I also met same error when I register consul with /v1/agent/service/register api, I found registered service just locate some consul server nodes not all server nodes
![image](https://user-images.githubusercontent.com/18571465/71714370-8d37f880-2e48-11ea-9edb-cf35ae17575f.png)
